cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -388,18 +388,21 @@ private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
                                                         ANNO_STRIMZI_IO_REBALANCE,
                                                         rawRebalanceAnnotation(updatedKafkaRebalance));
                                                 if (hasRebalanceAnnotation(updatedKafkaRebalance)) {
-                                                    LOGGER.debugCr(reconciliation, "Removing annotation {}={}",
-                                                            ANNO_STRIMZI_IO_REBALANCE,
-                                                            rawRebalanceAnnotation(updatedKafkaRebalance));
-                                                    // Updated KafkaRebalance has rebalance annotation removed as
-                                                    // action specified by user has been completed.
-                                                    KafkaRebalance patchedKafkaRebalance = new KafkaRebalanceBuilder(updatedKafkaRebalance)
-                                                            .editMetadata()
+                                                    if (rebalanceAnnotation.equals(KafkaRebalanceAnnotation.unknown)) {
+                                                        return kafkaRebalanceOperator.patchAsync(reconciliation, updatedKafkaRebalance);
```

Comment:
we are leaving annotation no why are we patching? any change to the KR resource?

Reply:
This is a good point, can you not just return `updatedKafkaRebalance` (or whatever results in essentially a no-op)?

Reply:
Resolved